### ISSUE TITLE
Feature request for "Guest Permission Level"

### ISSUE THREAD

**Issue by derkan** [go-gitea/gitea#4093]
## Description

This a **feature request** for defining a new permission level named as `Guest`. Rights for this level will be:

- Add, view Issues (Close only his/her issue)
- View Wiki (no updating, deleting)
- Watch/Unwatch Project
- Star/Unstart Project
- Will be able to view and download from releases
- Will not be able to view or update private repository code and files 
- Will not be able to view or update private repository pull requests
- Will not be able to fork, create repository, migration or organization

Actually `Guest` permission level will be a restricted version of `Read` level without any privileges on private repository codes/files.

This permission is important to make internal or external users to add/watch issues to private repositories themselves.

Please check [here](https://docs.gitlab.com/ee/user/permissions.html). As far as I see, Gitea needs `Guest` permission level. By this permission level Gitea;

- Can be used for documentation(wiki), 
- Binary distribution(releases) 
- Bug tracking(issues)

for **PRIVATE** repositories.

A small comparison:

|Gitea|GitLab|
|------|-------|
|Admin|Owner/Maintainer|
|Read|Reporter|
|Write|Developer|
|!!!|Guest|

**Comment by daviian** [comment]
I think this is already possible with org teams and unit type permissions.

**Comment by lafriks** [comment]
Not quite, as you can not specify write to issues but read only for other stuff

**Comment by derkan** [comment]
I've checked Gitlab. And released that what I want is `Guest` permission level. Please check [here](https://docs.gitlab.com/ee/user/permissions.html). I think Gitlab is also using static permission levels(no way to define new levels).  As far as I see, Gitea needs `Guest` permission level.(not reporter, I'll rename title).

A small comparison:

|Gitea|GitLab|
|------|-------|
|Admin|Owner/Maintainer|
|Read|Reporter|
|Write|Developer|
|!!!|Guest|


Regards

PS: Edited namings similar to Gitlab's, and updated issue content and this message's content for this change.

**Comment by daviian** [comment]
@lafriks AFAIK a read only team is able to create issues at the moment

So a read-only team with disabled Code and Pull Requests has exactly the permissions as requested.
The team members are able to see/create issues, see releases and read the wiki.

**Comment by 6543** [comment]
@derkan is https://github.com/go-gitea/gitea/pull/6274 what you wanted?

**Comment by derkan** [comment]
@6543 thanks for ping back. This PR brings a use case: A company runs a Gitea instance that requires login. Most repos are public (accessible/browsable by all co-workers). At some point, a customer or third party needs access to a specific repo and only that repo. Making such a customer account restricted and granting any needed access using team membership(s) and/or collaboration(s) is a simple way to achieve that without the need to make everything private.

But in my feature request this case is needed:  A company runs a Gitea instance and wants to allow public users to read wiki,add/read his issuses, download releases for a repo if rights given to him. By this we can use gitea for:
- Binary distribution(automatic updates with using release files)
- Online help (wiki)
- Issue handling
Currently, we can not limit user for not accessing source codes(and PR's)
